### PR TITLE
Tags to default spans

### PR DESCRIPTION
**Pull request by Shairam**
## Purpose
 Addition of tags to the default spans that are started automatically when the resources are serving to a request.

## Goals
Ability to include user defined tags to the default spans by passing -1 as the span id argument to the addTagtoSpan function as mentioned below.

> _ = observe:addTagToSpan(-1, "tot_requests",  <string> req_count);
 

## Approach
 Checks if  a default span is created when a resource starts serving and if it is present, the observercontext is returned and the tag is added to it.

### CODE REVIEW

**Comment by Anoukh** [comment]
@Shairam Can you reword the commit messages according to the standard we follow https://chris.beams.io/posts/git-commit/.

File: stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java [hunk 1]
```diff
@@ -139,16 +140,25 @@ public boolean finishSpan(int spanId) {
...
-    public boolean addTag(String tagKey, String tagValue, int spanId) {
+    public boolean addTag(String tagKey, String tagValue, int spanId, Context context) {
         if (!enabled) {
             return false;
         }
         ObserverContext observerContext = observerContextList.get(spanId);
-        if (observerContext != null) {
+        if (spanId == -1) {
+            Optional<ObserverContext> obs =   ObservabilityUtils.getParentContext(context);
+            if (obs.isPresent()){
```
Comment (Anoukh) [inline stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java]: There are few checkstyle issues. Can you fix them, run a local build and update the PR?

Comment (Shairam) [reply stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java]: okay ayiaa i will see to that..

Comment (Shairam) [reply stdlib/observability/src/main/java/org/ballerinalang/observe/nativeimpl/OpenTracerBallerinaWrapper.java]: Sure I will check on that

**Pull request closed without merging**